### PR TITLE
[telemetry certs] deploy certs for telemetry in deploy-mg

### PR DESCRIPTION
**Pull request by pra-moh**
### Description of PR
Adding server and dsmsroot certs to start telemetry in auth mode

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### How did you verify/test it?
using deploy-mg on virtual test bed 
image

### CODE REVIEW

File: ansible/config_sonic_basedon_testbed.yml [hunk 1]
```diff
@@ -35,6 +35,56 @@
...
   - block:
+    - name: Creates telemetry directory
+      file:
+          path: /etc/sonic/telemetry
+          state: directory
+      become: true
+
+    - name: Create telemetry server private key
+      openssl_privatekey:
+          path: /etc/sonic/telemetry/streamingtelemetryserver.key
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed.yml:46]: can you define var and then reuse var in your code?

File: ansible/config_sonic_basedon_testbed.yml [hunk 2]
```diff
@@ -35,6 +35,56 @@
...
+          privatekey_path: /etc/sonic/telemetry/streamingtelemetryserver.key
+          csr_path: /etc/sonic/telemetry/streamingtelemetryserver.csr
+          subject:
+              commonName: ndastreamingservertest.osdinfra.net
+          provider: selfsigned
+      become: true
+
+    - name: Create telemetry dsmsroot private key
+      openssl_privatekey:
+          path: /etc/sonic/telemetry/dsmsroot.key
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed.yml:68]: define vars

File: ansible/config_sonic_basedon_testbed.yml [hunk 3]
```diff
@@ -35,6 +35,56 @@
...
+      openssl_csr:
+          path: /etc/sonic/telemetry/streamingtelemetryserver.csr
+          privatekey_path: /etc/sonic/telemetry/streamingtelemetryserver.key
+      become: true
+    
+    - name: Generate a Self Signed OpenSSL telemetry server certificate
+      openssl_certificate:
+          path: /etc/sonic/telemetry/streamingtelemetryserver.cer
+          privatekey_path: /etc/sonic/telemetry/streamingtelemetryserver.key
+          csr_path: /etc/sonic/telemetry/streamingtelemetryserver.csr
```
Comment (hui-ma) [inline ansible/config_sonic_basedon_testbed.yml:60]: What is the relationship of "create telemetry server csr" and "Generate a Self Signed OpenSSL telemetry server certificate"?  Are the .csr and .key used as the input of the latter?

Comment (hui-ma) [inline ansible/config_sonic_basedon_testbed.yml:60]: why do you need csr and cer both? Can cer file be created directly?

File: ansible/config_sonic_basedon_testbed.yml [hunk 4]
```diff
@@ -35,6 +35,56 @@
...
+          privatekey_path: /etc/sonic/telemetry/dsmsroot.key
+      become: true
+    
+    - name: Generate a Self Signed OpenSSL telemetry dsmsroot certificate
+      openssl_certificate:
+          path: /etc/sonic/telemetry/dsmsroot.cer
+          privatekey_path: /etc/sonic/telemetry/dsmsroot.key
+          csr_path: /etc/sonic/telemetry/dsmsroot.csr
+          subject:
+              commonName: ndastreamingservertest.osdinfra.net
```
Comment (hui-ma) [inline ansible/config_sonic_basedon_testbed.yml:84]: I think this should be ndastreamingclienttest.osdinfra.net.

File: ansible/config_sonic_basedon_testbed.yml [hunk 5]
```diff
@@ -35,6 +35,56 @@
...
+    - name: Create telemetry dsmsroot private key
+      openssl_privatekey:
+          path: /etc/sonic/telemetry/dsmsroot.key
+          size: 2048
+      become: true
+   
+    - name: create telemetry dsmsroot csr
+      openssl_csr:
+          path: /etc/sonic/telemetry/dsmsroot.csr
+          privatekey_path: /etc/sonic/telemetry/dsmsroot.key
```
Comment (hui-ma) [inline ansible/config_sonic_basedon_testbed.yml:75]: dsmsroot should be the root cert of client side? Therefore, in the test case, it should be the client cert itself. Will this cert be used as telemetry client and copied to the test VM running the pytest?

File: ansible/config_sonic_basedon_testbed.yml (hunk 3 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:60]: no csr_path is required attribute self-signed for openssl_certificate https://docs.ansible.com/ansible/latest/modules/openssl_certificate_module.html

Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:60]: Yes. .csr and .key ar einput parameters for openssl_certificate

File: ansible/config_sonic_basedon_testbed.yml (hunk 5 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:75]: Yes. Plan is to use this cert on client-side. Do you think of any blockers?

File: ansible/config_sonic_basedon_testbed.yml (hunk 4 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:84]: Sure I will change that.

File: ansible/config_sonic_basedon_testbed.yml (hunk 5 above)
Comment (hui-ma) [reply ansible/config_sonic_basedon_testbed.yml:75]: No blocker. Agree with you and just want to check if what is your plan.

File: ansible/config_sonic_basedon_testbed.yml (hunk 1 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml]: Updated

File: ansible/config_sonic_basedon_testbed.yml (hunk 2 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml]: Updated

File: ansible/config_sonic_basedon_testbed.yml [hunk 6]
```diff
@@ -35,6 +35,81 @@
...
+
+    - name: create telemetry server csr
+      openssl_csr:
+          path: "{{ server_csr }}"
+          privatekey_path: "{{ server_key }}"
+      become: true
+    
+    - name: Generate a Self Signed OpenSSL telemetry server certificate
+      openssl_certificate:
+          path: /etc/sonic/telemetry/streamingtelemetryserver.cer
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed.yml:85]: why not use variable for this one as other ones?

File: ansible/config_sonic_basedon_testbed.yml [hunk 7]
```diff
@@ -35,6 +35,81 @@
...
+   
+    - name: create telemetry dsmsroot csr
+      openssl_csr:
+          path: "{{ dsmsroot_csr }}"
+          privatekey_path: "{{ dsmsroot_key }}"
+      become: true
+    
+    - name: Generate a Self Signed OpenSSL telemetry dsmsroot certificate
+      openssl_certificate:
+          path: /etc/sonic/telemetry/dsmsroot.cer
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed.yml:105]: why not use variable for this one as other ones?

File: ansible/config_sonic_basedon_testbed.yml [hunk 8]
```diff
@@ -35,6 +35,81 @@
...
+    - name: read dsmsroot csr
+      set_fact:
+          dsmsroot_csr: "{{ telemetry_certs['dsmsroot_csr'] }}"
+      when: telemetry_certs['dsmsroot_csr'] is defined
+
+    - name: Create telemetry server private key
+      openssl_privatekey:
+          path: "{{ server_key }}"
+          size: 2048
+      become: true
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed.yml:75]: should the telemtry server private key on the dut? should this key go to the ptf docker, where the server will be?

File: ansible/config_sonic_basedon_testbed.yml (hunk 6 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:85]: This one has not been reused. Do we need a variable in that case?

File: ansible/config_sonic_basedon_testbed.yml (hunk 8 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:75]: The telemetry server private key is needed on the server side. On ptf docker client side cert will be needed which in this case will be "dsmsroot.cer" basically plan is to use this cert on ptfdocker side . Any better suggestions?

Comment (lguohan) [reply ansible/config_sonic_basedon_testbed.yml:75]: which on is the server, the sonic or the ptf docker?

Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:75]: sonic is the server and ptfdocker will have gnmi_get(client side)

Comment (hui-ma) [reply ansible/config_sonic_basedon_testbed.yml:75]: How will this dsmsroot.cer and ky be copied onto ptf docker? And you will need renaming as well. Will it be handled in this PR or another PR?

Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:93]: added support to copy on ptf host

File: ansible/config_sonic_basedon_testbed.yml (hunk 6 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml]: added variable

File: ansible/config_sonic_basedon_testbed.yml (hunk 7 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml]: added variable

File: ansible/config_sonic_basedon_testbed.yml (hunk 8 above)
Comment (pra-moh) [reply ansible/config_sonic_basedon_testbed.yml:93]: added support to copy client certs on ptf docker

**Pull request merged**